atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/async/CqlClientImpl.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.cassandra.async;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.Executor;
+
+import com.datastax.driver.core.BoundStatement;
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Session;
+import com.google.common.collect.Streams;
+import com.google.common.util.concurrent.AsyncFunction;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.palantir.async.initializer.AsyncInitializer;
+import com.palantir.logsafe.Preconditions;
+
+@SuppressWarnings("checkstyle:FinalClass")
+public class CqlClientImpl implements CqlClient {
+    private static final class InitializingWrapper extends AsyncInitializer implements AutoDelegate_CqlClient {
+        private final Cluster cluster;
+        private final Executor executor;
+        private CqlClientImpl internalImpl;
+
+        InitializingWrapper(Cluster cluster, Executor executor) {
+            this.cluster = cluster;
+            this.executor = executor;
+        }
+
+        @Override
+        public CqlClient delegate() {
+            checkInitialized();
+            return internalImpl;
+        }
+
+        @Override
+        public CqlQueryBuilder<Object> queryBuilder() {
+            return internalImpl.queryBuilder();
+        }
+
+        @Override
+        protected void tryInitialize() {
+            internalImpl = new CqlClientImpl(cluster.connect(), executor);
+        }
+
+        @Override
+        protected String getInitializingClassName() {
+            return "CqlClient";
+        }
+
+        @Override
+        public void close() {
+            if (internalImpl != null) {
+                internalImpl.close();
+            }
+        }
+    }
+
+    private final Session session;
+    private final Executor executor;
+
+    public static CqlClient create(Cluster cluster,
+            Executor executor, boolean initializeAsync) {
```

Comment:
nit: all the same line, or all on individual lines please

Reply:
Same line.